concrete-core/src/lib.rs

```diff
@@ -1,316 +1,44 @@
 //!
-//! If you are not accustomed to cryptography, but are still interested by performing you should
-//! check the [`concrete`](https://crates.io/crates/concrete) library, which provides a simpler,
-//! higher-level API.
+//! # Audience
 //!
-//! # Quick Example
+//! This library is geared towards people who already know their way around Fhe. If you are not a
+//! cryptographer, this library may be unsafe/unsecure for you to use, as it requires you to tune
+//! a breadth of parameters which have an impact on the security/speed trade-off.
 //!
-//! Despite being low-overhead, `concrete-core` offers a pretty straightforward interface:
-//! ```
-//! // This examples shows how to multiply a secret value by a public one homomorphically. First
-//! // we import the proper symbols:
-//! use concrete_commons::dispersion::LogStandardDev;
-//! use concrete_commons::parameters::LweDimension;
-//! use concrete_core::crypto::encoding::{Cleartext, Encoder, Plaintext, RealEncoder};
-//! use concrete_core::crypto::lwe::LweCiphertext;
-//! use concrete_core::crypto::secret::generators::{
-//!     EncryptionRandomGenerator, SecretRandomGenerator,
-//! };
-//! use concrete_core::crypto::secret::LweSecretKey;
+//! Hopefully, we propose multiple libraries that build on top of `concrete-core` and which propose
+//! a safer/secure API. To see which one best suits your needs, see the
+//! [concrete homepage](https://zama.ai/concrete).
 //!
-//! // We initialize a prng that will be used to generate secret keys.
-//! let mut secret_generator = SecretRandomGenerator::new(None);
-//! // We initialize a prng used to encrypt data.
-//! let mut encryption_generator = EncryptionRandomGenerator::new(None);
-//! // We initialize an encoder that will allow us to turn cleartext values into plaintexts.
-//! let encoder = RealEncoder {
-//!     offset: 0.,
-//!     delta: 100.,
-//! };
-//! // Our secret value will be 10.,
-//! let cleartext = Cleartext(10_f64);
-//! let public_multiplier = Cleartext(5);
-//! // We encode our cleartext
-//! let plaintext = encoder.encode(cleartext);
+//! # Architecture
 //!
-//! // We generate a new secret key which is used to encrypt the message
-//! let secret_key_size = LweDimension(710);
-//! let secret_key = LweSecretKey::generate_binary(secret_key_size, &mut secret_generator);
+//! `concrete-core` is a modular library which makes it possible to use different backends to
+//! perform fhe operations. Its design revolves around two modules:
 //!
-//! // We allocate a ciphertext and encrypt the plaintext with a secure parameter
-//! let mut ciphertext = LweCiphertext::allocate(0u32, secret_key_size.to_lwe_size());
-//! secret_key.encrypt_lwe(
-//!     &mut ciphertext,
-//!     &plaintext,
-//!     LogStandardDev::from_log_standard_dev(-17.),
-//!     &mut encryption_generator,
-//! );
+//! + The [`specification`] module contains a specification (in the form of traits) of the concrete
```

Comment:
I was thinking `concrete` as the library. I made it clearer.

Reply:
Actually there are several places where "the concrete FHE scheme is used". The more I read it the clearer it appeared that it was the library we're talking about. So maybe there's no need to update this sentence (or maybe we should update every sentence that uses this).

Reply:
I just replaced `concrete scheme` by `concrete library scheme` for now. Does it suit you if I do this every time ? Btw, I don't use the concrete word for any other meaning than `concrete library` in the docs, because it is too confusing otherwise. Maybe it should be considered ? Or maybe I can turn `concrete` into a link to the zama page ?!

Reply:
Is it possible to replace concrete by `concrete`? This might actually be enough to do the trick.

Reply:
Absolutely !